packages/web/src/common/store/pages/audio-rewards/sagas.ts

```diff
@@ -320,25 +309,14 @@ function* claimSingleChallengeRewardAsync(
     )
     yield* put(setUserChallengesDisbursed({ challengeId, specifiers: claimed }))
 
-    // Ignore other errors and return early if there are AAO errors.
-    const aaoErrors = results.filter(
-      (r): r is AAOErrorResult => 'aaoErrorCode' in r
-    )
-    if (aaoErrors.length > 0) {
-      for (const error of aaoErrors) {
-        console.error(
-          `AAO attestation failed for specifier: ${error.specifier} for amount: ${error.amount}`
-        )
-      }
-      // Only consider the first AAO error code; others are likely to be the same anyway for the same user.
-      const aaoErrorCode = aaoErrors.map((r) => r.aaoErrorCode)[0]
-      return { aaoErrorCode }
-    }
-
     const errors = results.filter((r): r is ErrorResult => 'error' in r)
+    let aaoError: Errors.AntiAbuseOracleError | undefined
     if (errors.length > 0) {
       // Log and report errors for each specifier that failed to claim
       for (const res of errors) {
+        if (!aaoError && res.error instanceof Errors.AntiAbuseOracleError) {
+          aaoError = res.error
```

Comment:
The anti abuse attestation success isn't based on challenge ID/specifier as far as I can remember, so maybe we just throw here right away? And maybe we don't care about the message "Some specifiers failed to claim" and can just rethrow the error?

Reply:
kind of is based on challenge id. there is a world where certain challenges may be disabled in aao. we also have a few rules that use the challenge id. also, doesnt hurt to get the error logs for aao attestation failure in sentry

Reply:
Fair enough. It is kinda interesting that the code here only takes and reports the first AAO error then though